src/fork_gc.c

```diff
@@ -526,17 +526,25 @@ static void resetCardinality(NumGcInfo *info, NumericRange *range, size_t blocks
   size_t startIdx = InvertedIndex_NumBlocks(range->entries) - blocksSinceFork; // Here `blocksSinceFork` > 0
   IndexBlock *startBlock = InvertedIndex_BlockRef(range->entries, startIdx);
   t_docId startId = IndexBlock_FirstId(startBlock);
-  QueryIterator *iter = NewInvIndIterator_NumericFull(range->entries);
-  // Skip to the starting ID
-  IteratorStatus status = iter->SkipTo(iter, startId);
+  IndexDecoderCtx decoderCtx = {.tag = IndexDecoderCtx_None};
+  IndexReader *reader = NewIndexReader(range->entries, decoderCtx);
+  RSIndexResult *res = NewNumericResult();
+  // TODO: use IndexReader_Seek once we switched to the Rust implementation
+  // as the new version has a seek implementation for all decoders.
+  bool reading = IndexReader_SkipTo(reader, startId) && IndexReader_Next(reader, res);
```

Comment:
Why do we need `&& IndexReader_Next(reader, res)`? We should handle a single result if there is one, no? Or am I missing something in the API

Reply:
Since we know there is a non-empty block with `startId`, I think we can avoid the second check.
Also, `bool reading` will always be `true` initially for the same reason, but for the sake of readability, we can keep it set to the `IndexReader_SkipTo` return value

Reply:
Got it.
We still need to call `IndexReader_Next()` before entering the `while` loop, so what about this? I think it's clearer to init `reading` with the `Next` return value then.